### PR TITLE
Allow reading Excel files from stream.

### PR DESCRIPTION
**Pull request by eveneveneven**
I wanted to update #252 originally created by @marc-lopez but I felt it was cleaner to start over than merging 2 years of changes. Please let me know if there is some clever way I've could've done this to keep the original PR.

### CODE REVIEW

File: FileHelpers.ExcelNPOIStorage/ExcelNPOIStorage.cs [hunk 1]
```diff
@@ -364,6 +366,26 @@ public override object[] ExtractRecords()
             if (String.IsNullOrEmpty(FileName))
                 throw new ExcelBadUsageException("You need to specify the WorkBookFile of the ExcelDataLink.");
 
+            return TryGetRecordsFromWorkbook(() => OpenWorkbook(FileName));
```
Comment (mcavigelli) [inline FileHelpers.ExcelNPOIStorage/ExcelNPOIStorage.cs]: Here we have two return statements: I assume the second one should be removed, correct?

Comment (eveneveneven) [reply FileHelpers.ExcelNPOIStorage/ExcelNPOIStorage.cs]: Ooops, although the code is unreachable that's a bit messy. Fixed.

**Comment by mcavigelli** [comment]
Hej Even

Thank you for your contribution.

Matthias

**Comment by eveneveneven** [comment]
No problem Matthias, I'm glad to contribute! Thanks for merging 😃

**Pull request merged**